utils/removeDiscordRoleFromUser.ts

```diff
@@ -0,0 +1,57 @@
+import { logType } from "../constants/logs";
+import discordActions from "../models/discordactions";
+import { addLog } from "../models/logs";
+import discordServices from "../services/discordService";
+import { userData } from "../types/global";
+
+/**
+ * Removes a Discord role from a user using Discord Id.
+ *
+ * @param {Object} userData - User data.
+ * @param {string} discordId - User's Discord ID.
+ * @param {string} roleId - Discord Role ID.
+ *
+ * @returns {Promise<{ success: boolean; message: string }>} - Result with success status and message.
+ */
+export const removeDiscordRoleFromUser = async (
+  userData: userData,
+  discordId: string,
+  roleId: string
+): Promise<{ success: boolean; message: string }> => {
+  try {
+    const { roleExists } = await discordActions.isGroupRoleExists({ roleid: roleId, rolename: null });
+
+    if (!roleExists) {
+      const message = "Role doesn't exist";
+      await addLog(logType.REMOVE_ROLE_FROM_USER_FAILED, { roleId }, { message, userid: discordId, userData });
+      throw new Error(message);
+    }
+
+    try {
+      await discordServices.removeRoleFromUser(roleId, discordId, userData);
+    } catch (error) {
+      const message = "Role deletion from discord failed";
+      await addLog(logType.REMOVE_ROLE_FROM_USER_FAILED, { roleId, userid: discordId, userData }, { message });
+      throw new Error(message);
+    }
+
+    const deleteResponse = await discordActions.removeMemberGroup(roleId, discordId);
+    if (deleteResponse && !deleteResponse.wasSuccess) {
+      const message = "Role deletion from database failed";
+      await addLog(logType.REMOVE_ROLE_FROM_USER_FAILED, { roleId, userid: discordId }, { message, userData });
+      throw new Error(message);
+    }
```

Comment:
What should happen if the role is removed from discord but the db operation failed? Shouldn't it role back the previous operation?

Reply:
Role back means? Again adding the role to the user in the discord?

Reply:
The mentioned situation occur right? How would you handle that?

Reply:
I think rolling back won't be a good strategy because the primary job of the function is to remove the role from discord. For the rare cases (maybe data inconsistency) for users whose role is not getting removed from db there needs manual intervention to look out for the actual root cause for which logs are being maintained. 
Also for this feature rolling back would make it problematic because the user won't be able to authorize again and remove the unverified role because the modal won't open as in_discord: true will already been done through addorupdate function in controller.

Reply:
What will happen if we reverse the operation?
Let's first delete it from db and then discord.

Reply:
What will be the purpose of reversing the operation?

Reply:
Wouldn't the whole feature gets blocked if the db deletion process fails?
From user perspective getting the role removed from discord is the first priority, shouldn't it be independent of the db deletion process?
Don't you think getting the role deleted from db(which is a low priority operation) should happen after role getting removed from discord?

Reply:
Have we agreed on the priority of operation? If yes, then it make sense.